### PR TITLE
Adding handler and binding when services.xml has <search>-field

### PR DESCRIPTION
**Pull request by henrhoi**
Adding GUIHandler and bindings */querybuilder/*, when services.xml contains <search>

### CODE REVIEW

File: config-model/src/main/java/com/yahoo/vespa/model/container/xml/ContainerModelBuilder.java [hunk 1]
```diff
@@ -668,6 +668,18 @@ private void addSearchHandler(ContainerCluster cluster, Element searchElement) {
         }
 
         cluster.addComponent(searchHandler);
+        addGUIHandler(cluster, searchElement);
+    }
+
+    private void addGUIHandler(ContainerCluster cluster, Element searchElement) {
+        ProcessingHandler<SearchChains> guiHandler = new ProcessingHandler<>(
+                cluster.getSearch().getChains(), "com.yahoo.search.query.gui.GUIHandler");
```
Comment (bratseth) [inline config-model/src/main/java/com/yahoo/vespa/model/container/xml/ContainerModelBuilder.java]: Do you really take the search chains as argument?

Comment (bratseth) [reply config-model/src/main/java/com/yahoo/vespa/model/container/xml/ContainerModelBuilder.java]: And I don't think your handler is a ProcessingHandler subclass?

**Review by bratseth — CHANGES_REQUESTED** [review]

**Review by bratseth — APPROVED** [review]

File: container-integration-test/src/test/java/com/yahoo/search/query/gui/GUIHandlerTest.java [hunk 2]
```diff
@@ -67,7 +66,7 @@ private void assertResponse(Request.Method method, String path, String expectedS
         Assert.assertEquals("Status code", expectedStatusCode, response.getStatus());
         Assert.assertEquals(expectedContentType, response.getHeaders().getFirst("Content-Type"));
         if(expectedStartString != null){
-            assertTrue(response.getBodyAsString().startsWith(expectedStartString));
+            Assert.assertTrue(response.getBodyAsString().startsWith(expectedStartString));
```
Comment (gjoranv) [inline container-integration-test/src/test/java/com/yahoo/search/query/gui/GUIHandlerTest.java]: Please keep the static import.

File: container-search-gui/src/main/java/com/yahoo/search/query/gui/GUIHandler.java [hunk 3]
```diff
@@ -47,71 +43,80 @@ public HttpResponse handle(HttpRequest request) {
...
+            if (path.endsWith(".css")) {
                 return "text/css";
-            } else if (path.toString().endsWith(".js")) {
+            } else if (path.endsWith(".js")) {
                 return "application/javascript";
-            } else if (path.toString().endsWith(".html")) {
+            } else if (path.endsWith(".html")) {
                 return "text/html";
-            }else if (path.toString().endsWith(".php")) {
+            }else if (path.endsWith(".php")) {
```
Comment (gjoranv) [inline container-search-gui/src/main/java/com/yahoo/search/query/gui/GUIHandler.java]: The code is a little easier to read if white-spaces are used consistently: `} else if (...) {

**Review by gjoranv — COMMENTED** [review]
Please consider my (optional) comments.

File: container-search-gui/src/main/java/com/yahoo/search/query/gui/GUIHandler.java (hunk 3 above)
Comment (henrhoi) [reply container-search-gui/src/main/java/com/yahoo/search/query/gui/GUIHandler.java]: I agree. I will go through it and correct it.

**Comment by gjoranv** [comment]
👍

**Comment by henrhoi** [comment]
@gjoranv I cant see why the CI-test failed after changing the "typo" in container-disc/pom.xml

**Comment by henrhoi** [comment]
The branch builds successfully local on my machine..

**Comment by gjoranv** [comment]
It's unrelated. Unit tests in 'jrt' are unstable also on my local builds.

Hint: If you sign up to travis, you can restart the build without modifying any files.

**Comment by henrhoi** [comment]
Okey, thanks!

**Pull request merged**